src/preamble/02_what_is_a_module_and_a_package.md

```diff
@@ -1 +1,366 @@
-# What is a module and a package?
+# What is a module, a component, and a package?
+
+## TL;DR
+
+There is an ongoing initiative to standardize the terminology we use to refer to
+the hierarchy of collections of code in the Haskell community. If you are
+interested in reading more about the you can find a more thorough treatment
+[here](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/compiler/units#background).
+
+The most significant concepts in this chapter all refer to ways of dividing and
+aggregating code. They form a hierarchy, and are somewhat defined in terms of
+each other. Here is a list of them in order from smallest to largest are:
+
+1. **Module** - The smallest unit of code defined by GHC and the Haskell
+   language specification.
+2. **Component** - A component is a collection of modules. Usually a `library`,
+   `executable`, or a `test-suite`.
+3. **Unit** - Units are compiled components, while not technically _larger_ than
+   a component, they are further along in the packaging process and therefore
+   closer to a distributable artefact.
+4. **Package** - A package is a collection of components. In the context of
+   cabal a package is identified by a `<package-name>.cabal` file. A package is
+   **the** unit of distribution in the Haskell ecosystem i.e. everything on
+   hackage is a package.
+5. **Project** - A project is a grouping of several related packages. In the
+   context of cabal this is denoted by a `cabal.project` file in the root
+   directory.
+
+Units and projects are not covered in this chapter, although projects have their
+own [chapter](../getting_fancy/01_setting_up_a_cabal_project.md)
```

Comment:
This seems to me a good organisation of material and what I've written above is mostly nitpicks. Will review the latter parts after I review this starting portion again.